File: 27-06-2025_Longest_Subsequence_Repeated_K_Times.java
public String longestSubsequenceRepeatedK(String s, int k) {
    String result = "";
    Queue<String> q = new LinkedList<>();
    q.offer("");

    while (!q.isEmpty()) {
        String curr = q.poll();
        for (char ch = 'a'; ch <= 'z'; ch++) {
            String next = curr + ch;
            if (isK(next, s, k)) {
                if (next.length() > result.length() || 
                   (next.length() == result.length() && next.compareTo(result) > 0)) {
                    result = next;
                }
                q.offer(next);
            }
        }
    }

    return result;
}

private boolean isK(String sub, String t, int k) {
    int count = 0, i = 0;
    for (char ch : t.toCharArray()) {
        if (ch == sub.charAt(i)) {
            i++;
            if (i == sub.length()) {
                i = 0;
                count++;
                if (count == k) return true;
            }
        }
    }
    return false;
}

File: 27-06-2025_Longest_Subsequence_Repeated_K_Times.py
def longestSubsequenceRepeatedK(self, s: str, k: int) -> str:
    from collections import deque

    def isK(sub: str, t: str, k: int) -> bool:
        count = 0
        i = 0
        for ch in t:
            if ch == sub[i]:
                i += 1
                if i == len(sub):
                    i = 0
                    count += 1
                    if count == k:
                        return True
        return False

    result = ""
    q = deque()
    q.append("")

    while q:
        curr = q.popleft()
        for ch in map(chr, range(ord('a'), ord('z') + 1)):
            next_str = curr + ch
            if isK(next_str, s, k):
                if (len(next_str) > len(result)) or \
                   (len(next_str) == len(result) and next_str > result):
                    result = next_str
                q.append(next_str)

    return result
